lib/blocktree/blocktree.go

```diff
@@ -69,6 +71,37 @@ func (bt *BlockTree) GenesisHash() Hash {
 	return bt.head.hash
 }
 
+// GetAllHashesForParentDepth will get all hashes with the same dept for a given header
+func (bt *BlockTree) GetAllHashesForParentDepth(header *types.Header) (map[common.Hash]*big.Int, error) {
+	hashes := map[common.Hash]*big.Int{}
+
+	parent := bt.getNode(header.ParentHash)
+	if parent == nil {
+		err := fmt.Errorf("cannot find parent block in blocktree")
+		log.Error("Could not getNode for ParentHash", "header.ParentHash", header.ParentHash, "err", err)
+		return hashes, err
+	}
+
+	parentDepthPlusOne := big.NewInt(0).Add(parent.depth, big.NewInt(1))
+
+	for _, child := range bt.head.children {
+		log.Debug("going to iterate ", "child.depth", child.depth.Int64(),
+			"parent.depth", parent.depth.Int64(),
+			"child.depth.Cmp(parent.depth)", child.depth.Cmp(parent.depth),
+			"child.hash.String()", child.hash.String(),
+			"hashes", len(hashes),
+			"bt.head.children", len(bt.head.children))
+
+		if child.depth.Cmp(parentDepthPlusOne) == 0 {
+			if hashes[child.hash] == nil {
+				log.Debug("adding child.hash", "child.hash", child.hash.String(), "child.depth", child.depth)
+				hashes[child.hash] = child.depth
+			}
+		}
+	}
+	return hashes, nil
```

Comment:
I don't think this function returns all nodes with the same depth. there might be multiple branches of the tree with nodes with the same depth, the nodes that are children of the parent arent necessarily all the nodes at that depth.

also, I'm not sure why you need to return a map. I think an array of hashes is sufficient

Reply:
I am returning a map to avoid duplicated results, since using a array would potentially cause duplicated results...

Reply:
there shouldn't be any duplicated results, since you can't add a block to the blocktree if a block with that hash is already in it